universal-login-commons/lib/integration/ethereum/resolveName.ts

```diff
@@ -10,9 +11,11 @@ export const resolveName = async (
   const node = utils.namehash(ensName);
```

Comment:
In general it is a bad practice to return complex type like `string | false`;
It is better to return a pair e.g.:
```ts
class EnsResolutionResult {
  exist: boolean;
  address: string;
}

const resolveName = async (...): EnsResolutionResult => {
...
}
```

or return nullable:
```ts
type Nullable<T> = T | null; //In commons/types

const resolveName = async (...): Nullable<string> => {
...
}

```

Reply:
We could use that opportunity to convert resolveName to service, the simplest option would be:
```ts
export const resolveName = (provider: providers.Provider, ensAddress: string) => 
 async (ensName: string) => {	
  ...
}
```


So that we can inject that with DI. Should make test a bit nicer.

Reply:
For current pr updated resolveName signature